lib/modules/manager/pub/artifacts.spec.ts

```diff
@@ -0,0 +1,156 @@
+import _fs from 'fs-extra';
+import { join } from 'upath';
+import { envMock, mockExecAll } from '../../../../test/exec-util';
+import { mocked } from '../../../../test/util';
+import { GlobalConfig } from '../../../config/global';
+import type { RepoGlobalConfig } from '../../../config/types';
+import * as _env from '../../../util/exec/env';
+import type { UpdateArtifactsConfig } from '../types';
+import * as pub from './artifacts';
+
+jest.mock('fs-extra');
+jest.mock('../../../util/exec/env');
+jest.mock('../../../util/git');
+jest.mock('../../../util/http');
+
+const fs: jest.Mocked<typeof _fs> = _fs as any;
+const env = mocked(_env);
+
+const adminConfig: RepoGlobalConfig = {
+  localDir: join('/tmp/github/some/repo'),
+  cacheDir: join('/tmp/cache'),
+  containerbaseDir: join('/tmp/cache/containerbase'),
+};
+
+const config: UpdateArtifactsConfig = {};
+
+describe('modules/manager/pub/artifacts', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+    jest.resetModules();
+
+    env.getChildProcessEnv.mockReturnValue(envMock.basic);
+    GlobalConfig.set(adminConfig);
+  });
+
+  afterEach(() => {
+    GlobalConfig.set(adminConfig);
+  });
+
+  it('returns null if no pubspec.lock found', async () => {
+    const updatedDeps = [{ depName: 'dep1' }];
+    expect(
+      await pub.updateArtifacts({
+        packageFileName: 'pubspec.yaml',
+        updatedDeps,
+        newPackageFileContent: '',
+        config,
+      })
+    ).toBeNull();
+  });
+
+  it('returns null if updatedDeps is empty', async () => {
+    expect(
+      await pub.updateArtifacts({
+        packageFileName: 'pubspec.yaml',
+        updatedDeps: [],
+        newPackageFileContent: '',
+        config,
+      })
+    ).toBeNull();
+  });
+
+  it('returns null if unchanged', async () => {
+    const execSnapshots = mockExecAll();
+    fs.readFile.mockResolvedValueOnce('Current pubspec.lock' as any);
+    fs.readFile.mockResolvedValueOnce('Current pubspec.lock' as any);
+    const updatedDeps = [{ depName: 'dep1' }];
+    expect(
+      await pub.updateArtifacts({
+        packageFileName: 'pubspec.yaml',
+        updatedDeps,
+        newPackageFileContent: '',
+        config,
+      })
+    ).toBeNull();
+    expect(execSnapshots).toMatchSnapshot();
+  });
+
+  it('returns updated dart pubspec.lock', async () => {
+    const execSnapshots = mockExecAll();
+    fs.readFile.mockResolvedValueOnce('Old pubspec.lock' as any);
+    fs.readFile.mockResolvedValueOnce('New pubspec.lock' as any);
+    const updatedDeps = [{ depName: 'dep1' }];
+    expect(
+      await pub.updateArtifacts({
+        packageFileName: 'pubspec.yaml',
+        updatedDeps,
+        newPackageFileContent: '',
+        config,
+      })
+    ).not.toBeNull();
```

Comment:
use toEqual

Reply:
jest complains that it should be toBeNull

Reply:
no. match the returned value explicit with `toEqual` instead of `not.toBeNull`.

Reply:
sorry i don't get it. are you saying it should be `.not.toEqual(null)` instead of `.not.toBeNull()`?